### DISCUSSION TITLE
How to show file particular path only on unfocused windows

### DISCUSSION THREAD

**Discussion by XXiaoA** [Bekaboo/dropbar.nvim#60] [Q&A] [answered]
Assuming  that we have three windows in a tab: `~/a` `~/b/file` `~/c/file`,  we focus on the windows `~/a` and its winbar show symbol as normal. But I want other two window show `b/file` and `c/file` in winbar instead of showing symbol.  How can we do that?

**Comment by Bekaboo** [comment]
You can create your own sources by wrapping existing sources and let the custom sources return empty lua table when the cursor is not at the window the dropbar instance is attached to.

**Reply by XXiaoA** [reply]
Thx! But I'm confused about how to change the bar's source when we are about to leave a window. Before entering a new window, we ought to update the current winbar to only one source `particular_path`, I think.

**Reply by XXiaoA** [reply]
Now I can create a source to show different content between focused window and unfocused windows.
```lua
    local source = {
        get_symbols = function(buf, win, cursor)
            if vim.api.nvim_get_current_win() ~= win then
                return {
                    bar.dropbar_symbol_t:new({
                        name = "Unfocused",
                        name_hl = "Title",
                    }),
                }
            end
            return { 
                    bar.dropbar_symbol_t:new({
                        name = "Focused",
                        name_hl = "Title",
                    }),
                 }

        end,
    }
 ```
Then we can set `opt.bar.sources = { source }`

The problem what I met is that in a focused window, we have loads of sources like `path` `lsp`. But in unfocused window, we have one source `p_path` only. I don't know how to do it. I tried use a function as `opts.bar.sources`'s value, like
```lua
sources = function(buf, win)
    if win == current_win then
         return { path, lsp }
    else
         return { p_path }
    end
end
```
But it doesn't work. Because  function is triggered when a new winbar is created only. Do u have any idea? @Bekaboo

**Reply by Bekaboo** [reply, marked as answer]
@XXiaoA You should create a source that returns an empty table when the source is requested to return symbols for a window that is not the current window, instead of trying to change the source when you switch the window.

**Reply by XXiaoA** [reply]
okay, i got it. Thx!

**Reply by iam-vipin** [reply]
Can you share the code in here ? 
@XXiaoA

**Reply by XXiaoA** [reply]
> Can you share the code in here ? 
> @XXiaoA 
> 

I have quited coding for a long time due to my study. Sorry
